pages/api/destinations/[id].js

```diff
@@ -0,0 +1,38 @@
+import {
+  deleteDestination,
+  getDestinationById,
+  updateDestination,
+} from "../../../services/destinationService";
+
+export default async function handler(req, res) {
+  const {
+    query: { id, tripId },
+    method,
+  } = req;
+
+  switch (method) {
+    case "GET":
+      const destination = await getDestinationById(id);
+      if (destination.error) return res.status(destination.status).json({ error: destination.error });
+      res.status(200).json(destination);
+      break;
+
+    case "PATCH":
+      const newDestination = await updateDestination(id, req.body);
+      if (newDestination.error)
+        return res.status(newDestination.status).json({ error: newDestination.error });
+      res.status(201).json(newDestination);
+      break;
+
+    case "DELETE":
+      const deletedDestination = await deleteDestination(id, tripId);
```

Comment:
The tripId should not be needed for deleting a destination. The tripId is just a field in the destination.

Reply:
done